### ISSUE TITLE
Question :  Best practice to iterate over multiple polygons 

### ISSUE THREAD

**Issue by kostastheocharoulis** [anitagraser/movingpandas#107]
Hi Amanda , 

Assume there is a polygon-area which is is splitted to smaller ones. 
![image](https://user-images.githubusercontent.com/79842137/113002537-8c406480-917a-11eb-8e2d-e5cedec1b782.png)

I am trying to find the most effective approach (from performance perspective) to calculate for each sub-polygon the trajectories which clip  to the corresponding polygons. 
![image](https://user-images.githubusercontent.com/79842137/113003997-eb52a900-917b-11eb-9b1b-1c9d7fbcf0d8.png).

I was thinking and I 've started implementing the approach of iterating in each polygon and clip the trajectories to each polygon, but it takes too long . 

So I would like to ask you based on your exprerience if there is another way to iplement it. 

Thanks in advance 
Kostas

**Comment by anitagraser** [comment]
Do you need clip or is a check for intersects enough? Are the trajectories sampled densely enough to just perform a point in polygon check for the original point locations?

**Comment by kostastheocharoulis** [comment]
Thanks Anita for the quick response . 

1) Regarding your first question the only thing I need to know is the start and the end time of the trajectory for each sub-polygon and the answer to this question in my opinion  is related to second one . i ll explain below

2)Let's assume the worst case scenario which may appear in my data, where only few points are located in the line-trajectory ( ** I have marked them with brown color below) . 
![image](https://user-images.githubusercontent.com/79842137/113062940-32ad5980-91bd-11eb-8438-215b8a5b2f36.png)

From what I understand traj-clip will calculate for each subpolygon the start and the end time  even there is n't any point in the polygon while intersect requires a point in the polygon to work the same , Am i right ?

**Comment by anitagraser** [comment]
> From what I understand traj-clip will calculate for each subpolygon the start and the end time even there is n't any point in the polygon while intersect requires a point in the polygon to work the same , Am i right ?

Not quite. Clip and intersect should behave in the same way. By default, they are line-based, i.e. no point inside the polygon is needed as long as the line goes through the polygon. This behavior can be changed by setting the `point_based=True` parameter, as documented in https://movingpandas.readthedocs.io/en/master/trajectory.html#movingpandas.Trajectory.clip and https://movingpandas.readthedocs.io/en/master/trajectory.html#movingpandas.Trajectory.intersection
